src/wp-includes/query.php

```diff
@@ -1117,7 +1118,20 @@ function _find_post_by_old_slug( $post_type ) {
 		$query .= $wpdb->prepare( ' AND DAYOFMONTH(post_date) = %d', get_query_var( 'day' ) );
 	}
 
-	$id = (int) $wpdb->get_var( $query );
+	$key          = md5( $query );
+	$last_changed = wp_cache_get( 'last_changed', 'posts' );
+	if ( ! $last_changed ) {
+		$last_changed = microtime();
+		wp_cache_set( 'last_changed', $last_changed, 'posts' );
+	}
```

Comment:
```suggestion
	$last_changed = wp_cache_get_last_changed( 'posts' );
```

:1234: Helper function can be used throughout.